DQM/Physics/src/SusyDQM.cc

```diff
@@ -0,0 +1,987 @@
+        for(uint jIndex = 0; jIndex < j_temp1.size(); jIndex++){
+            thisSumEt1 += j_temp1[jIndex].Pt();
+        }
+        double thisSumEt2 = 0;
+        for(uint jIndex = 0; jIndex < j_temp2.size(); jIndex++){
+            thisSumEt2 += j_temp2[jIndex].Pt();
+        }
+        double thisDeltaHT = fabs(thisSumEt1 - thisSumEt2);
+        if(minDeltaHT < 0 || thisDeltaHT < minDeltaHT) minDeltaHT = thisDeltaHT;
+    }
+    return minDeltaHT;
+}
+
+//adapted from JetMET/plugins/HLTRFilter.cc
+template <typename Mu, typename Ele, typename Pho, typename Jet, typename Met>
+double SusyDQM<Mu, Ele, Pho, Jet, Met>::calcMR(TLorentzVector ja, TLorentzVector jb) {
+    if(ja.Pt()<=0.2) return -999;
+    ja.SetPtEtaPhiM(ja.Pt(),ja.Eta(),ja.Phi(),0.0);
+    jb.SetPtEtaPhiM(jb.Pt(),jb.Eta(),jb.Phi(),0.0);
+    if(ja.Pt() > jb.Pt()){
+        TLorentzVector temp = ja;
+        ja = jb;
+        jb = temp;
+    }
+    double A = ja.P();
+    double B = jb.P();
+    double az = ja.Pz();
+    double bz = jb.Pz();
+    TVector3 jaT, jbT;
+    jaT.SetXYZ(ja.Px(),ja.Py(),0.0);
+    jbT.SetXYZ(jb.Px(),jb.Py(),0.0);
+    double ATBT = (jaT+jbT).Mag2();
+    double MR = sqrt((A+B)*(A+B)-(az+bz)*(az+bz)-
+            (jbT.Dot(jbT)-jaT.Dot(jaT))*(jbT.Dot(jbT)-jaT.Dot(jaT))/(jaT+jbT).Mag2());
+    double mybeta = (jbT.Dot(jbT)-jaT.Dot(jaT))/
+        sqrt(ATBT*((A+B)*(A+B)-(az+bz)*(az+bz)));
+    double mygamma = 1./sqrt(1.-mybeta*mybeta);
+    //use gamma times MRstar
+    return MR*mygamma;
+}
+
+template <typename Mu, typename Ele, typename Pho, typename Jet, typename Met>
+double SusyDQM<Mu, Ele, Pho, Jet, Met>::calcRsq(double MR, TLorentzVector ja, TLorentzVector jb, edm::Handle<std::vector<Met> > inputMet) {
+    TVector3 met;
+    met.SetPtEtaPhi((inputMet->front()).pt(),0.0,(inputMet->front()).phi());
+    double MTR = sqrt(0.5*(met.Mag()*(ja.Pt()+jb.Pt()) - met.Dot(ja.Vect()+jb.Vect())));
+    double R = MTR/MR;
+    return R*R;
+}
+
+template <typename Mu, typename Ele, typename Pho, typename Jet, typename Met>
+double SusyDQM<Mu, Ele, Pho, Jet, Met>::calcMT2(float testMass, bool massive, vector<TLorentzVector> jets, TLorentzVector MET, int hemi_seed, int hemi_association){
+    //computes MT2 using a test mass of testMass, with hemispheres made massless if massive is set to false
+    //hemispheres are clustered by finding the grouping of input jets that minimizes the Lund distance
+
+    using namespace dqm;
+
+    if(jets.size() < 2) return -9999; //need at least two jets for the calculation
+    vector<float> px, py, pz, E;
+    for(uint i = 0; i < jets.size(); i++){
+        //push 4vector components onto individual lists
+        px.push_back(jets[i].Px());
+        py.push_back(jets[i].Py());
+        pz.push_back(jets[i].Pz());
+        E.push_back(jets[i].E());
+    }
+    
+    //form the hemispheres using the provided Hemisphere class
+    Hemisphere* hemis = new Hemisphere(px, py, pz, E, hemi_seed, hemi_association);
```

Comment:
If it is a temporary object, why not put it into a scope (and worry less about lifetime/deletion)?
`Hemisphere hemis(px, py, pz, E, hemi_seed, hemi_association);`